src/logic/config/store/reducer/index.ts

```diff
@@ -0,0 +1,31 @@
+import { Action, handleActions } from 'redux-actions'
+
+import { AppReduxState } from 'src/store'
+import { CONFIG_ACTIONS } from 'src/logic/config/store/actions'
+import { makeNetworkConfigFromSpecificConfiguration, NetworkState } from 'src/logic/config/model/networkConfig'
+import { getConfig, getNetworkId, getNetworkName, NetworkSpecificConfiguration, setNetworkId } from 'src/config'
+import { ETHEREUM_NETWORK } from 'src/config/networks/network.d'
+
+export const NETWORK_CONFIG_REDUCER_ID = 'networkConfig'
+
+type Payload = NetworkState | NetworkSpecificConfiguration | string
+
+export default handleActions<AppReduxState['networkConfig'], Payload>(
+  {
+    [CONFIG_ACTIONS.CONFIG_STORE]: (state, action: Action<NetworkSpecificConfiguration>) => {
+      const prevState = { ...state }
+      const newParams = makeNetworkConfigFromSpecificConfiguration(action.payload)
+      return { ...prevState, ...newParams }
+    },
+    [CONFIG_ACTIONS.CONFIG_SET_CHAIN_ID]: (state, action: Action<string>) => {
+      const prevState = { ...state }
+      setNetworkId(getNetworkName(action.payload as ETHEREUM_NETWORK))
```

Comment:
the reducer shouldn't contain any side effects, also the network id name is a little misleading because we're getting a name?

Reply:
This was done right now because it's a bit hacky. We are triggering this action from Redux Dev tools, we still don't have a button on where to trigger all this that should be the place where it should be done

For convenience it was easier to pass the chainId, and also it's the value we will get from the config service. But the static config relies on a networkName value to be initialized, as it needs to load the file with the correct network name